lib/chef/knife/core/subcommand_loader.rb

```diff
@@ -28,8 +28,8 @@ class SubcommandLoader
       attr_reader :chef_config_dir
       attr_reader :env
 
-      def initialize(chef_config_dir, env=ENV)
```

Comment:
it's impossible to know if folks are using that argument really. Looking back at 9239ff893ea5b36db03ab53489376dcbd9536ecd, I don't see that it was ever really used, maybe @danielsdeleo remembers more.

I'd consider something like

```
def initialize(chef_config_dir, env=nil)
  if env.nil?
    env=ENV
  else
    Chef::Log.deprecation("The env argument to Chef::Knife::SubcommandLoader is deprecated, please file an issue at https://github.com/chef/chef/issues if you're using it.")
  end
end
```